src/components/instantlaunches/index.js

```diff
@@ -0,0 +1,151 @@
+                {t("submittingInstantLaunch")}
+            </DialogTitle>
+
+            <DialogContent>
+                <div
+                    className={classes.progress}
+                    id={buildID(baseID, ids.PROGRESS)}
+                >
+                    <LinearProgress />
+                </div>
+
+                <Typography variant="h6" className={classes.inProgress}>
+                    {t("submissionInProgress")}
+                </Typography>
+
+                <Typography variant="body2" className={classes.closeAuto}>
+                    {t("dialogWillCloseAutomatically")}
+                </Typography>
+            </DialogContent>
+        </Dialog>
+    );
+};
+
+/**
+ * An IconButton with a handler that launches the app in the instant launch with
+ * the resource as an input.
+ *
+ * @param {Object} props
+ * @param {Object} props.instantLaunch - The instant launch to use.
+ * @param {Object} props.resource - The resource to use as an input to the instant launch.
+ */
+const InstantLaunchButton = ({
+    instantLaunch,
+    resource = {},
+    showErrorAnnouncer,
+    size = "medium",
+    themeSpacing = 0,
+    color = "primary",
+}) => {
+    const baseID = buildID(ids.BASE, ids.LAUNCH, ids.BUTTON);
+    const router = useRouter();
+    const [open, setOpen] = React.useState(false);
+    const theme = useTheme();
+
+    return (
+        <IconButton
+            id={baseID}
+            variant="contained"
+            size={size}
+            style={{ marginLeft: theme.spacing(themeSpacing) }}
+            color={color}
+            onClick={async (event) => {
+                event.stopPropagation();
+                event.preventDefault();
+
+                setOpen(true);
+
+                await instantlyLaunch(instantLaunch, resource)
+                    .then((listing) => {
+                        if (listing.analyses.length > 0) {
+                            const analysis = listing.analyses[0];
+                            if (analysis.interactive_urls.length > 0) {
+                                router
+                                    .push(
+                                        `${
+                                            constants.VICE_LOADING_PAGE
+                                        }/${encodeURIComponent(
+                                            analysis.interactive_urls[0]
+                                        )}`
+                                    )
+                                    .then(() => setOpen(false));
+                            } else {
+                                setOpen(false);
+                            }
+                        } else {
+                            setOpen(false);
+                        }
+                    })
+                    .catch((error) => {
+                        setOpen(false);
+                        showErrorAnnouncer(error.message, error);
+                    });
+            }}
+        >
```

Comment:
I was curious so I tried this myself, and the below should work.  You'll also have to add the `useMutation` import at the top though.
 
```suggestion

    const [instantLaunchMutation] = useMutation(instantlyLaunch, {
        onSuccess: (listing) => {
            if (listing.analyses.length > 0) {
                const analysis = listing.analyses[0];
                if (analysis.interactive_urls.length > 0) {
                    router
                        .push(
                            `${
                                constants.VICE_LOADING_PAGE
                            }/${encodeURIComponent(
                                analysis.interactive_urls[0]
                            )}`
                        )
                        .then(() => setOpen(false));
                } else {
                    setOpen(false);
                }
            } else {
                setOpen(false);
            }
        },
        onError: (error) => {
            setOpen(false);
            showErrorAnnouncer(error.message, error);
        }
    });

    return (
        <IconButton
            id={baseID}
            variant="contained"
            size={size}
            style={{ marginLeft: theme.spacing(themeSpacing) }}
            color={color}
            onClick={(event) => {
                setOpen(true)
                instantLaunchMutation(instantLaunch, resource)
            }}
        >
```

Reply:
I'll work on this in a separate PR after the merge. The stuff that's there is working and this PR is already too large.

Reply:
I filed https://cyverse.atlassian.net/browse/CORE-1396 to track the Formik usage refactor.